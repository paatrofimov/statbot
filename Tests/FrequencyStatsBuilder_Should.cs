using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using StatBot.Vk.Models;
using StatBot.Vk.Requests;
using StatBot.VkApiClient.Models.Internal;

namespace Tests
{
    [TestFixture]
    public class FrequencyStatsBuilder_Should
    {
        private FrequencyStatsBuilder builder;

        [OneTimeSetUp]
        public void OneTimeSetUp()
        {
            builder = new FrequencyStatsBuilder();
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void BuildFrequency()
        {
            var result = builder.Build(new StatsBuilderBuildArgs
                                       {
                                           Username = "Username",
                                           WallGetObject = new InternalVkApiWallGetObject
                                                           {
                                                               WallApiPosts = new[]
                                                                              {
                                                                                  new InternalVkApiPost
                                                                                  {
                                                                                      Text =
                                                                                          "ПоcтрОение таблицы чАстоТ символов"
                                                                                  }
                                                                              }
                                                           }
                                       });

            Approvals.Verify(result);
        }
    }
}