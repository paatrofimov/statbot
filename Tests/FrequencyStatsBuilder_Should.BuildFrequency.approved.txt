Username, статистика для последних 1 постов: {п:0,03,о:0,16,c:0,03,т:0,13,р:0,03,е:0,06,н:0,03,и:0,10,а:0,06,б:0,03,л:0,06,ц:0,03,ы:0,03,ч:0,03,с:0,06,м:0,03,в:0,06}